pkg/service/service.go

```diff
@@ -28,62 +29,28 @@ import (
 
-// ProviderIndex is a read/write interface to a local cache of providers that falls back to IPNI
-type ProviderIndex interface {
-	// Find should do the following
-	//  1. Read from the IPNI Storage cache to get a list of providers
-	//     a. If there is no record in cache, query IPNI, filter out any non-content claims metadata, and store
-	//     the resulting records in the cache
-	//     b. the are no records in the cache or IPNI, it can attempt to read from legacy systems -- Dynamo tables & content claims storage, synthetically constructing provider results
-	//  2. With returned provider results, filter additionally for claim type. If space dids are set, calculate an encodedcontextid's by hashing space DID and Hash, and filter for a matching context id
-	//     Future TODO: kick off a conversion task to update the recrds
-	Find(context.Context, providerindex.QueryKey) ([]model.ProviderResult, error)
-	// Cache writes entries to the cache but does not publish/announce an
-	// advertisement for them. Entries expire after a pre-determined time.
-	Cache(ctx context.Context, provider peer.AddrInfo, contextID string, digests []multihash.Multihash, meta meta.Metadata) error
-	// Publish should do the following:
-	// 1. Write the entries to the cache with no expiration until publishing is complete
-	// 2. Generate an advertisement for the advertised hashes and publish/announce it
-	Publish(ctx context.Context, provider peer.AddrInfo, contextID string, digests []multihash.Multihash, meta meta.Metadata) error
-}
-
-// ClaimLookup is used to get full claims from a claim cid
-type ClaimLookup interface {
-	// LookupClaim should:
-	// 1. attempt to read the claim from the cache from the encoded contextID
-	// 2. if not found, attempt to fetch the claim from the provided URL. Store the result in cache
-	// 3. return the claim
-	LookupClaim(ctx context.Context, claimCid cid.Cid, fetchURL url.URL) (delegation.Delegation, error)
-}
-
-// BlobIndexLookup is a read through cache for fetching blob indexes
-type BlobIndexLookup interface {
-	// Find should:
-	// 1. attempt to read the sharded dag index from the cache from the encoded contextID
-	// 2. if not found, attempt to fetch the index from the provided URL. Store the result in cache
-	// 3. return the index
-	// 4. asyncronously, add records to the ProviderStore from the parsed blob index so that we can avoid future queries to IPNI for
-	// other multihashes in the index
-	Find(ctx context.Context, contextID types.EncodedContextID, provider model.ProviderResult, fetchURL url.URL, rng *metadata.Range) (blobindex.ShardedDagIndexView, error)
-}
-
 // IndexingService implements read/write logic for indexing data with IPNI, content claims, sharded dag indexes, and a cache layer
 type IndexingService struct {
-	blobIndexLookup BlobIndexLookup
-	claimLookup     ClaimLookup
-	providerIndex   ProviderIndex
+	blobIndexLookup blobindexlookup.BlobIndexLookup
```

Comment:
Apologies for my GO related thinking here -- there's a meme in go of declare the interface where you use it. Honestly, not sure I've ever fully bought in :)

Reply:
![i_have_no_idea_what_im_doing](https://github.com/user-attachments/assets/19039d6e-e664-44db-81b8-6b6035790fe2)